Task4_AWSCloud Formation.yaml

```diff
@@ -0,0 +1,212 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: >
+  CloudFormation template for Networking and Subnets to create a VPC with public and private subnets in multiple availability zones.
+
+Parameters:
+  VpcCidr:
+    Type: String
+    Default: 172.16.0.0/16
+    Description: CIDR block for the VPC
+  PublicSubnetCidr1:
+    Type: String
+    Default: 172.16.1.0/24
+    Description: CIDR block for Public Subnet 1
+  PublicSubnetCidr2:
+    Type: String
+    Default: 172.16.2.0/24
+    Description: CIDR block for Public Subnet 2
+  PrivateSubnetCidr1:
+    Type: String
+    Default: 172.16.3.0/24
+    Description: CIDR block for Private Subnet 1
+  PrivateSubnetCidr2:
+    Type: String
+    Default: 172.16.4.0/24
+    Description: CIDR block for Private Subnet 2
+
+Resources:
+  #VPC && InternetGateway.
+  #---- VPC Creation.
+  VPC:
+    Type: AWS::EC2::VPC
+    Properties:
+      CidrBlock: !Ref VpcCidr
+      EnableDnsSupport: true
+      EnableDnsHostnames: true
+      Tags:
+        - Key: Name
+          Value: DevVPC
+  #---- InternetGateway Creation.
+  InternetGateway:
+    Type: AWS::EC2::InternetGateway
+    Properties:
+      Tags:
+        - Key: Name
+          Value: InternetGatewayForDevVPC
+  #---- AttachGateway To VPC.
+  InternetGatewayAttachment:
+    Type: AWS::EC2::VPCGatewayAttachment
+    Properties:
+      VpcId: !Ref VPC
+      InternetGatewayId: !Ref InternetGateway
+    DependsOn:
+      - InternetGateway
+      - VPC
+  # Elastic IPs for NAT Gateways
+  NatEip1:
+    Type: AWS::EC2::EIP
+    Properties:

```

Comment:
You need to add Name Tag, example:
```
      Tags:
        - Key: Name
          Value: YYYY
```

Reply:
Done
![image](https://github.com/user-attachments/assets/760477ec-c1e8-49f5-bd12-35857100dbda)